File: integration-tests/gradle/src/main/resources/inject-quarkus-app-properties/src/main/java/org/acme/ExampleResource.java
package org.acme;

import jakarta.inject.Inject;
import jakarta.ws.rs.GET;
import jakarta.ws.rs.Path;
import jakarta.ws.rs.Produces;
import jakarta.ws.rs.core.MediaType;

import org.eclipse.microprofile.config.inject.ConfigProperty;

@Path("/hello")
public class ExampleResource {

    @ConfigProperty(name = "my-app-name")
    String appName;
    @ConfigProperty(name = "quarkus.application.version")
    String appVersion;

    @GET
    @Produces(MediaType.TEXT_PLAIN)
    public String hello() {
        return appName + " " + appVersion;
    }
}

File: integration-tests/gradle/src/main/resources/kotlin-grpc-project/src/main/kotlin/org/acme/ExampleResource.kt
package org.acme

import jakarta.ws.rs.GET
import jakarta.ws.rs.Path
import jakarta.ws.rs.Produces
import jakarta.ws.rs.core.MediaType

import io.quarkus.example.HelloMsg
import io.quarkus.example.helloMsg

@Path("/hello")
class ExampleResource {

    @GET
    @Produces(MediaType.TEXT_PLAIN)
    fun hello() = "hello" + helloMsg { status = HelloMsg.Status.TEST_ONE }.statusValue
}